### PR TITLE
Add Message model

### LINKED ISSUE #14: Add Messages Page
Depends on #6

Every pet  should have a Messages page

- UI with a list of messages from other pets should be created
- Probably a Message controller shall be needed

### PR DESCRIPTION
**Pull request by c-d**
Resolves #14 - Add Message model and EF migrations.

### CODE REVIEW

File: AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs [hunk 1]
```diff
@@ -37,6 +39,20 @@ protected override void OnModelCreating(ModelBuilder builder)
                 .HasForeignKey(like => like.LikedPetId)
                 .OnDelete(DeleteBehavior.Restrict);
 
+            builder
+                .Entity<Pet>()
+                .HasMany(pet => pet.SentMessages)
+                .WithOne(message => message.Sender)
+                .HasForeignKey(message => message.RecipientId)
```
Comment (tcholakov) [inline AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs]: I think the RecipientId should be a foreign key for the Recepient and should be linked to ReceivedMessages in Pet

File: AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs [hunk 2]
```diff
@@ -37,6 +39,20 @@ protected override void OnModelCreating(ModelBuilder builder)
...
+                .OnDelete(DeleteBehavior.Restrict);
+
+            builder
+                .Entity<Pet>()
+                .HasMany(pet => pet.ReceivedMessages)
+                .WithOne(message => message.Recipient)
+                .HasForeignKey(message => message.SenderId)
```
Comment (tcholakov) [inline AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs]: The SenderId should be a foreign key for the Sender and should be linked to SentMessages in Pet

File: AnimalMatcher/AnimalMatcher.Data/Models/Message.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        public Pet Recipient { get; set; }
+
+        [Required]
+        public int SenderId { get; set; }
+
+        public Pet Sender { get; set; }
+
+        [MaxLength(MessageConstants.SubjectMaxLength)]
+        public string MessageSubject { get; set; }
```
Comment (tcholakov) [inline AnimalMatcher/AnimalMatcher.Data/Models/Message.cs]: Messages will not need a Subject - as we plan to have a chat between two pets which will contain a list of messages between them and will have only a message body and a timestamp - when the message was sent

File: AnimalMatcher/AnimalMatcher.Data/Models/Message.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        
+        [MaxLength(MessageConstants.BodyMaxLength)]
+        public string MessageBody { get; set; }
```
Comment (tcholakov) [inline AnimalMatcher/AnimalMatcher.Data/Models/Message.cs]: [suggestion] Could be renamed to Body as it is a part of the Message class and it clearly relates to the Message

File: AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs (hunk 1 above)
Comment (c-d) [reply AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs]: Good point, updated with last commit.

File: AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs (hunk 2 above)
Comment (c-d) [reply AnimalMatcher/AnimalMatcher.Data/AnimalMatcherDbContext.cs]: Good point, updated with last commit.

File: AnimalMatcher/AnimalMatcher.Data/Models/Message.cs (hunk 3 above)
Comment (c-d) [reply AnimalMatcher/AnimalMatcher.Data/Models/Message.cs]: Removed subject now.

File: AnimalMatcher/AnimalMatcher.Data/Models/Message.cs (hunk 4 above)
Comment (c-d) [reply AnimalMatcher/AnimalMatcher.Data/Models/Message.cs]: Yep fair call, have corrected this now.

**Pull request merged**